File: csharp/TicketOfficeTest.cs
using NUnit.Framework;

namespace KataTrainReservation
{
    [TestFixture]
    public class TicketOfficeTest {
    
        [Test]
        public void reserveSeats() {
            TicketOffice office = new TicketOffice("http://localhost:8081", "http://localhost:8082");
            ReservationRequest request = new ReservationRequest("express_2000", 4);
    
            Reservation reservation = office.makeReservation(request);
    
            Assert.AreEqual(4, reservation.seats.length);
            Assert.AreEqual("A", reservation.seats[0].coach);
            Assert.AreEqual("75bcd15", reservation.booking_reference);

        }
    }
}

File: Java/src/test/java/TicketOfficeTest.java
import static org.junit.Assert.*;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.mock;

import org.junit.Test;
import org.mockito.Matchers;

public class TicketOfficeTest {

    @Test
    public void shouldCallTrainDataServiceForTrainInformation() {
        TrainDataService trainDataService = mock(TrainDataService.class);
        TicketOffice office = new TicketOffice(trainDataService, null);

        ReservationRequest request = new ReservationRequest("express_2000", 4);

        office.makeReservation(request);

        verify(trainDataService).getTrainInformation(Matchers.eq("express_2000"));
    }
}
